From 1fbc8d0bc7042ba2bef8fc25627e6fc9b6b82627 Mon Sep 17 00:00:00 2001
From: "Angelo G. Del Regno" <kholk11@gmail.com>
Date: Wed, 20 Mar 2019 13:02:36 +0100
Subject: [PATCH 021/178] media: camera_v2: csiphy: Vary CDR settings for
 SDM660

Following Qualcomm, vary the CSIPHY CDR regulator voltage to
run the PHY in CPHY mode and the CDR mask setting to match
the HPG on these chips.

Change-Id: Id9c8d0f97918f8b8481dd0d108ba8940543638db
---
 .../media/platform/msm/camera_v2/sensor/csiphy/msm_csiphy.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/drivers/media/platform/msm/camera_v2/sensor/csiphy/msm_csiphy.c b/drivers/media/platform/msm/camera_v2/sensor/csiphy/msm_csiphy.c
index d5af192953d5..7ed7be998a55 100644
--- a/drivers/media/platform/msm/camera_v2/sensor/csiphy/msm_csiphy.c
+++ b/drivers/media/platform/msm/camera_v2/sensor/csiphy/msm_csiphy.c
@@ -2480,8 +2480,12 @@ static int csiphy_probe(struct platform_device *pdev)
 		"qcom,csiphy-v3.5")) {
 		if (of_machine_is_compatible("qcom,sdm630") ||
 		    of_machine_is_compatible("qcom,sdm636") ||
-		    of_machine_is_compatible("qcom,sdm660"))
+		    of_machine_is_compatible("qcom,sdm660")) {
+			csiphy_v3_5_3ph.mipi_csiphy_3ph_lnn_ctrl23.data = 0x23;
+			csiphy_v3_5_3ph.mipi_csiphy_3ph_lnn_ctrl25.data = 0x50;
 			csiphy_v3_5_3ph.mipi_csiphy_3ph_lnn_ctrl26.data = 0x70;
+		}
+
 		new_csiphy_dev->ctrl_reg->csiphy_3ph_reg = csiphy_v3_5_3ph;
 		new_csiphy_dev->ctrl_reg->csiphy_reg = csiphy_v3_5;
 		new_csiphy_dev->hw_dts_version = CSIPHY_VERSION_V35;
-- 
2.49.0

